datalad/runner/nonasyncrunner.py

```diff
@@ -304,42 +303,46 @@ def run(self) -> Union[Any, Generator]:
             self.stdin_queue = Queue()
             self.stdin_queue.put(self.stdin)
             self.stdin_queue.put(None)
+
         elif isinstance(self.stdin, Queue):
             # Establish a queue to write to the process.
             self.write_stdin = True
             self.stdin_queue = self.stdin
+
         else:
-            # indicate that we will not write anything to stdin, that
-            # means the user can pass None, or he can pass a
-            # file-like and write to it from a different thread.
-            lgr.warning(f"Unknown instance class: {type(self.stdin)}, "
-                        f"assuming file-like input: {self.stdin}")
-            # We assume that the caller will write to the given
-            # file descriptor.
+            # We do not recognize the input class will and just pass is through
+            # to Popen(). We assume that the caller handles any writing if
+            # desired.
             self.write_stdin = False
 
         self.protocol = self.protocol_class(**self.protocol_kwargs)
 
+        # The following command is generated internally by datalad
+        # and trusted. Security check is therefore skipped.
         kwargs = {
             **self.popen_kwargs,
             **dict(
                 bufsize=0,
                 stdin=subprocess.PIPE if self.write_stdin else self.stdin,
                 stdout=subprocess.PIPE if self.catch_stdout else None,
                 stderr=subprocess.PIPE if self.catch_stderr else None,
-                shell=True if isinstance(self.cmd, str) else False
+                shell=True if isinstance(self.cmd, str) else False      # nosec
```

Comment:
What nosec - what understand it, code climate?

Reply:
It is understood by [bandit](https://github.com/PyCQA/bandit), a security linter that is used by codeclimate.